### ISSUE TITLE
Failed checks due to missing wazuh-indexer package

### ISSUE THREAD

**Issue by MarcelKemp** [wazuh/wazuh-packages#2749]
## Description 

PR #2746 checks are failing due to the following reasons:
- The `4.8` image generated for RPM is created from the branch of the following development (not yet merged): https://github.com/wazuh/wazuh/issues/14153
  - They will be fixed at the time the development is merged (although it could also be fixed by generating an image for GitHub without that development).

- The **`wazuh-indexer`** packages cannot be generated, which are causing many failures in the checks: `All in One` and `Distributed cases (indexer)`, because it does not find the package.
  - This case needs to be investigated to fix it, as it has been failing since previous upward PRs, and should have been investigated in:   https://github.com/wazuh/wazuh-packages/issues/2717
  - **Attempt to generate packages: https://ci.wazuh.info/job/Packages_builder_tier/3351/**

### Wazuh agent RPM packages build

Error:
```
+ /usr/local/bin/rpmbuild --define '_sysconfdir /etc' --define '_topdir /build_wazuh/rpmbuild' --define '_threads 2' --define '_release merge.4.8.0.into.4.8.1' --define '_localstatedir /var/ossec' --define '_debugenabled no' --define '_download_content_enabled ' --target x86_64 -ba /build_wazuh/rpmbuild/SPECS/wazuh-agent-4.8.1.spec
error: Macro %_download_content_enabled has empty body
```

The error appears with the build of the RPM package via GitHub action: 
- https://github.com/wazuh/wazuh-packages/actions/runs/7421515316/job/20195025439?pr=2746

### All in one and Distributed cases

Error (or similar):
```
No package wazuh-indexer-4.8.1 available.
...
05/01/2024 17:23:32 ERROR: Wazuh indexer installation failed.
```

All In One - Amazonlinux 2_base installation  — Test_unattended
- https://github.com/wazuh/wazuh-packages/pull/2746/checks?check_run_id=20205134425

All In One - Centos 7 installation  — Test_unattended
- https://github.com/wazuh/wazuh-packages/pull/2746/checks?check_run_id=20205126003

All In One - Centos 8 installation  — Test_unattended
- https://github.com/wazuh/wazuh-packages/pull/2746/checks?check_run_id=20205761848

All In One - Rhel 7 installation  — Test_unattended
- https://github.com/wazuh/wazuh-packages/pull/2746/checks?check_run_id=20205163173

All In One - Rhel 8 installation  — Test_unattended
- https://github.com/wazuh/wazuh-packages/pull/2746/checks?check_run_id=20205198961

All In One - Rhel 9 installation  — Test_unattended
- https://github.com/wazuh/wazuh-packages/pull/2746/checks?check_run_id=20205127885

All In One - Ubuntu bionic installation  — Test_unattended
- https://github.com/wazuh/wazuh-packages/pull/2746/checks?check_run_id=20205229899

All In One - Ubuntu focal installation  — Test_unattended
- https://github.com/wazuh/wazuh-packages/pull/2746/checks?check_run_id=20205181185

All In One - Ubuntu jammy installation  — Test_unattended
- https://github.com/wazuh/wazuh-packages/pull/2746/checks?check_run_id=20205189445

All In One - Ubuntu xenial installation  — Test_unattended
- https://github.com/wazuh/wazuh-packages/pull/2746/checks?check_run_id=20205096721

Distributed - Certificates  — Test_unattended_distributed
- https://github.com/wazuh/wazuh-packages/pull/2746/checks?check_run_id=20205504796

Distributed - Wazuh indexer installation  — Test_unattended_distributed
- https://github.com/wazuh/wazuh-packages/pull/2746/checks?check_run_id=20195758617

Distributed Cases - manager - centos - Wazuh indexer installation  — Test_unattended_distributed_cases
- https://github.com/wazuh/wazuh-packages/pull/2746/checks?check_run_id=20205155512

Distributed Cases - manager - ubuntu - Wazuh indexer installation  — Test_unattended_distributed_cases
- https://github.com/wazuh/wazuh-packages/pull/2746/checks?check_run_id=20205411877

Distributed Cases - manager_dashboard - centos - Wazuh indexer installation  — Test_unattended_distributed_cases
- https://github.com/wazuh/wazuh-packages/pull/2746/checks?check_run_id=20205221486

Distributed Cases - manager_dashboard - ubuntu - Wazuh indexer installation  — Test_unattended_distributed_cases
- https://github.com/wazuh/wazuh-packages/pull/2746/checks?check_run_id=20205247959

Distributed Cases - manager_indexer - centos - Wazuh indexer installation  — Test_unattended_distributed_cases
- https://github.com/wazuh/wazuh-packages/pull/2746/checks?check_run_id=20205409200

Distributed Cases - manager_indexer - ubuntu - Wazuh indexer installation  — Test_unattended_distributed_cases
- https://github.com/wazuh/wazuh-packages/pull/2746/checks?check_run_id=20205249941

## DoD

- [x] Find the cause why wazuh-indexer packages (version >= `4.8.1`) can't be generated correctly.
- [x] Fixing the problem to be able to generate the packages
- [x] Once generated, check by re-running checks

**Comment by vikman90** [comment]
Let's assign this issue to @wazuh/frontend.

**Comment by rauldpm** [comment]
###About this issue

This issue presents the following main errors:

1. RPM agent build error in GHA -> @wazuh/core
2. Wazuh installation assistant tests error -> @wazuh/cicd

From what I can see of the error in the tests, it is because the packages do not exist, which should be the responsibility of CICD as it is related to its tests, on the other hand, in the link to the package builder, is possible to see that it fails when building them due to the lack of the following TD resources

> Packages_rpm_indexer_x86_64_4-8-1
> Packages_deb_indexer_amd64_4-8-1
> Packages_deb_dashboard_amd64_4-8-1
> Packages_rpm_dashboard_x86_64_4-8-1

The creation of these Task Definitions is carried out (or should be) during upward merges by the @wazuh/core team, an example of the latest generation of resources (4.7.2) https://github.com/wazuh/internal-devel-requests/issues/558, this process should be included in some procedure such as merges or support for new patches

It is also possible to see an error when uploading a package due to an already-known error https://github.com/wazuh/wazuh-jenkins/issues/4639 whose solution is to repeat the build

- https://ci.wazuh.info/job/Packages_uploader_repository/11092/console

An issue should be opened for each team for each error cc @MarcelKemp (this issue could be edited and used for one of them), with an additional issue to create the missing ECS TD resources.

**Comment by jotacarma90** [comment]
## Update 15/01/2024
Images created:
```
C:\Users\Vhalgarv\Wazuh\wazuh-packages\wazuhapp\opensearch-dashboards\Docker (4.8.1 -> origin)
λ docker images
REPOSITORY                                                        TAG                                   IMAGE ID       CREATED          SIZE
966237403726.dkr.ecr.us-east-1.amazonaws.com/package-generation   dashboardapp_builder_noarch_4-8-1     2e03b866e528   30 seconds ago   647MB
966237403726.dkr.ecr.us-east-1.amazonaws.com/package-generation   deb_indexer_builder_amd64_4-8-1       6aa13c1a4f91   24 minutes ago   647MB
966237403726.dkr.ecr.us-east-1.amazonaws.com/package-generation   rpm_indexer_builder_x86_64_4-8-1      66c7b089c166   26 minutes ago   600MB
966237403726.dkr.ecr.us-east-1.amazonaws.com/package-generation   deb_dashboard_builder_amd64_4-8-1     ea6464a9c22c   28 minutes ago   972MB
966237403726.dkr.ecr.us-east-1.amazonaws.com/package-generation   rpm_dashboard_builder_x86_64_4-8-1    8a11c3fece53   2 hours ago      1.29GB
966237403726.dkr.ecr.us-east-1.amazonaws.com/package-generation   base_dashboard_builder_amd64_4-8-1    c2acc156e067   2 hours ago      875MB
966237403726.dkr.ecr.us-east-1.amazonaws.com/package-generation   base_dashboard_builder_x86_64_4-8-1   c2acc156e067   2 hours ago      875MB
966237403726.dkr.ecr.us-east-1.amazonaws.com/package-generation   base_indexer_builder_amd64_4-8-1      77b3c89d68f2   2 hours ago      918MB
966237403726.dkr.ecr.us-east-1.amazonaws.com/package-generation   base_indexer_builder_x86_64_4-8-1     77b3c89d68f2   2 hours ago      918MB
```

Images uploaded:
![image](https://github.com/wazuh/wazuh-packages/assets/60003131/f5d7e580-6f72-493e-a0f3-a8ba2fde5e11)


Creating task definitions for those images in progress...

**Comment by rauldpm** [comment]
If the AWS ECR/ECS resources have been created and the packages are available, I guess that yes, the issue can be closed, but are the PR checks still failing?

**Comment by jotacarma90** [comment]
## Update 18/01/2024

I have recreated the PR branch:
https://github.com/wazuh/wazuh-packages/pull/2746
To relaunch the checks, you need to create the indexer and dashboard packages in staging. I have launched this pipe for it:
https://ci.wazuh.info/job/Packages_builder_tier/3370

Once everything is ready, I will relaunch the checks and everything should be fine.

**Comment by jotacarma90** [comment]
## Update 23/01/2024
Issue unlocked, the problem that prevented generating the packages has been fixed. Builds of jenkins:
- https://ci.wazuh.info/job/Packages_builder/181857/
- https://ci.wazuh.info/job/Packages_builder/181858/

Updated ETA to today, everything should be finished.

**Comment by jotacarma90** [comment]
## Conclusion
Hello team, after some problems with failed checks, what I have done is to open a PR dummy modifying the same files that were modified in the PR we were testing:
- PR to test: https://github.com/wazuh/wazuh-packages/pull/2746/files
- PR dummy: https://github.com/wazuh/wazuh-packages/pull/2793/files

We believe that what was happening is that, since the PR was already merged, the PR checks were not detecting the changes in the branch, and they were still taking images of the moment in which those checks were launched for the first time. In short, now with the PR dummy all the checks are green for 4.8.1.

So this issue can be closed.

**Comment by MarcelKemp** [comment]
## Review 
The problems reported in the issue have been fixed, and all checks have been verified to pass correctly :green_circle: 
So LGTM!

GJ Jose!

**Comment by MarcelKemp** [comment]
We seem to have fixed version `4.8.1`, however `4.8.2` requires the same steps, so I am reopening the issue:
- https://github.com/wazuh/wazuh-packages/pull/2819
```
No package wazuh-indexer-4.8.2 available.
```
